### PR TITLE
CEDS-3033 Update test deps and tests refactoring

### PR DESCRIPTION
**Pull request by udinobi**

### CODE REVIEW

File: test/it/base/IntegrationSpec.scala [hunk 1]
```diff
@@ -1,6 +0,0 @@
-package base
-
-import org.scalatest.concurrent.{IntegrationPatience, ScalaFutures}
-import org.scalatest.{MustMatchers, WordSpec}
-
-trait IntegrationSpec extends WordSpec with MustMatchers with ScalaFutures with IntegrationPatience
```
Comment (MaciejRewera) [inline test/it/base/IntegrationSpec.scala:6]: 👍

File: test/unit/services/notifications/NotificationServiceSpec.scala [hunk 2]
```diff
@@ -18,22 +18,21 @@ package services.notifications
...
 import repositories.NotificationsRepository
 import testdata.notifications.ExampleXmlAndNotificationDetailsPair.exampleNotification
 import testdata.notifications.NotificationsTestData._
 
-import scala.concurrent.ExecutionContext.Implicits.global
-import scala.concurrent.Future
-import scala.xml.NodeSeq
-
-class NotificationServiceSpec extends UnitSpec with BeforeAndAfterEach with ScalaFutures {
+class NotificationServiceSpec extends UnitSpec with ScalaFutures {
```
Comment (MaciejRewera) [inline test/unit/services/notifications/NotificationServiceSpec.scala:35]: Aren't `ScalaFutures` already part of `UnitSpec` ?

File: test/unit/controllers/actions/AuthActionSpec.scala [hunk 3]
```diff
@@ -69,7 +68,7 @@ class AuthActionSpec extends PlaySpec with ScalaCheckPropertyChecks with Mockito
       "authorise returns an error" in {
 
         when(mockAuthConnector.authorise[Option[String]](any(), any())(any(), any()))
-          .thenReturn(Future.failed(new Exception()))
+          .thenReturn(Future.failed(new RuntimeException("")))
```
Comment (MaciejRewera) [inline test/unit/controllers/actions/AuthActionSpec.scala:71]: 👍

**Review by MaciejRewera — APPROVED** [review]
LGTM
Although, my approach is to extend as little as possible. But I am happy to make this service a playground for an opposite.

File: test/unit/services/notifications/NotificationServiceSpec.scala (hunk 2 above)
Comment (udinobi) [reply test/unit/services/notifications/NotificationServiceSpec.scala:35]: It was already removed but not committed yet :smile: 
I'll update it with the next PR.

**Pull request merged**